src/System.Net.Security/src/System/Net/NegotiateStreamPal.Windows.cs

```diff
@@ -0,0 +1,467 @@
+                Interop.SspiCli.Endianness.Network,
+                inSecurityBufferArray,
+                outSecurityBuffer,
+                ref outContextFlags);
+
+            contextFlags = ContextFlagsAdapterPal.GetContextFlagsPalFromInterop(outContextFlags);
+            return SecurityStatusAdapterPal.GetSecurityStatusPalFromInterop(winStatus);
+        }
+
+        internal static void ValidateImpersonationLevel(TokenImpersonationLevel impersonationLevel)
+        {
+            if (impersonationLevel != TokenImpersonationLevel.Identification &&
+                impersonationLevel != TokenImpersonationLevel.Impersonation &&
+                impersonationLevel != TokenImpersonationLevel.Delegation)
+            {
+                throw new ArgumentOutOfRangeException("impersonationLevel", impersonationLevel.ToString(), SR.net_auth_supported_impl_levels);
+            }
+        }
+
+        internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
+        {
+            return new Win32Exception((int)SecurityStatusAdapterPal.GetInteropFromSecurityStatusPal(statusCode));
+        }
+
+        internal static int Encrypt(
+            SafeDeleteContext securityContext,
+            byte[] buffer,
+            int offset,
+            int count,
+            bool isConfidential,
+            bool isNtlm,
+            ref byte[] output,
+            uint sequenceNumber)
+        {
+            SecSizes sizes = SSPIWrapper.QueryContextAttributes(
+                GlobalSSPI.SSPIAuth,
+                securityContext,
+                Interop.SspiCli.ContextAttribute.Sizes
+                ) as SecSizes;
+
+            try
+            {
+                int maxCount = checked(Int32.MaxValue - 4 - sizes.BlockSize - sizes.SecurityTrailer);
+
+                if (count > maxCount || count < 0)
+                {
+                    throw new ArgumentOutOfRangeException("count", SR.Format(SR.net_io_out_range, maxCount));
+                }
+            }
+            catch (Exception e)
+            {
+                if (!ExceptionCheck.IsFatal(e))
+                {
+                    if (GlobalLog.IsEnabled)
+                    {
+                        GlobalLog.Assert("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Encrypt", "Arguments out of range.");
+                    }
+
+                    Debug.Fail("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Encrypt", "Arguments out of range.");
+                }
+
+                throw;
+            }
+
+            int resultSize = count + sizes.SecurityTrailer + sizes.BlockSize;
+            if (output == null || output.Length < resultSize + 4)
+            {
+                output = new byte[resultSize + 4];
+            }
+
+            // Make a copy of user data for in-place encryption.
+            Buffer.BlockCopy(buffer, offset, output, 4 + sizes.SecurityTrailer, count);
```

Comment:
I'm probably just missing it, but...
Here we're writing count bytes starting at 4+sizes.SecurityTrailer.  And below we fill in the first 4 bytes.  But where do we fill in those sizes.SecurityTrailer bytes?  And resultSize includes space for sizes.BlockSize... does that space filled in somewhere?